static/js/containers/pages/DashboardPage.js

```diff
@@ -367,18 +428,32 @@ export class DashboardPage extends React.Component<Props, State> {
                   ))}
               </div>
             </div>
+            {currentUser.is_authenticated &&
+              (currentUser.unused_coupons.length > 0 ? (
+                <React.Fragment>
+                  <h3>Unredeemed Enrollment Code(s)</h3>
+                  <div className="enrollment-codes">
+                    {currentUser.unused_coupons.map(this.renderEnrollmentCode)}
```

Comment:
if I go with bulk enrollment and uploaded a user csv multiple times for the same course, will it not render this area again and again over the user dashboard ?

Reply:
I am not sure if understood your question correctly. I would say this PR does not alter any backend logic, it will simply render the list in `unused_coupon` returned by `api/users/me`

Reply:
earlier, I believe the system was sending the email but now we are updating the dashboard.  After looking around the dashboard_page changes, It seems to me, If a staff upload the same `user.csv` while performing the bulk enrollment again for the same course then will it not render the same block twice over the user dashboard ? 
please correct me over it if I am missing something over it.

Reply:
it shouldn't.

Reply:
Please give a try for final confirmation. thanks

Reply:
yeah sure, I'll do.

Reply:
@asadiqbal08 I have deep dive into the issue and here are my findings.

- Creating coupons: You can specify no. of coupons.
- You can assign each coupon code to one user at a time.
- You can specify multiple coupons of the same course to one user.

**Scenario I tested upon your request**

- Created 3 coupon codes.
- Created CSV with only one user.
- Uploaded it multiple times.
- Each time when we upload a new coupon code (if available) assigned to given user.
- Once each coupon code assigned to specific user, the admin will get an error while uploading it again.
- You can also download coupon assignment CSV, after uploading.
- `/api/users/me` will return unused coupon codes. 
- Above return results rendered on the dashboard three times see the screenshot attached, with different coupon codes.

![Screenshot 2019-12-09 at 4 22 24 PM](https://user-images.githubusercontent.com/4245618/70432704-b1025b80-1aa1-11ea-8afe-36db9c33a34f.png)

Let me know if I missed anything, I am happy to test every edge case here.

Reply:
So from UI prospective, I am not sure either is it good behavior or not to render a same block multiple times with multiple codes over the user dashboard. Personally, I will not prefer to see such stuff over my dashboard but ideally one block with one code for a course. 

Secondly, I just saw another thing, I uploaded a `users.csv` and enroll users for `Demo-course` then I upload the same `users.csv` and chose `SEED Models in Engineering` now I am seeing the same code for two different courses over a user dashboard. 

For ref: 
<img width="1272" alt="Screen Shot 2019-12-09 at 6 57 05 PM" src="https://user-images.githubusercontent.com/7334669/70441317-fb8dd300-1ab5-11ea-8273-c42b9f0a7f48.png">

Reply:
@pdpinch I would like to get you input on this before changing any behavior or UI.

- We can give one enrollment code to a user for different courses, the user will see both courses on the dashboard until he use one of them.
- We can give multiple coupons to one user, in that case, the user will see multiple entries on dashboard for the same course but with different enrollment codes.

If we want to change behavior in any of the above scenarios, we have to update our API which is returning `unused_coupons` for logged-in users.

Reply:
I'm not very clear on what you are asking here. 

1. "We can give one enrollment code to a user for different courses, the user will see both courses on the dashboard until he use one of them." 

Although enrollment codes can support multiple CouponEligibility objects, this is rarely used -- typically only for MIT staff and TAs. I think it's fine to show all the courses until the user uses the code (or the code expires). 

2. "We can give multiple coupons to one user, in that case, the user will see multiple entries on dashboard for the same course but with different enrollment codes." 

This is an edge case that I'm not too worried about. It will likely only affect MIT staff. 

That said, it is important that this feature in general be behind a feature flag, in case I discover that I'm wrong in my assumptions.